src/LinkContainer.js

```diff
@@ -1,46 +1,28 @@
-import React, { Component } from 'react';
+import React from 'react';
 import PropTypes from 'prop-types';
-import { Route, withRouter } from 'react-router-dom';
+import {
+  Route, useHref, useLocation, useMatch, useNavigate,
+} from 'react-router-dom';
+import { isFunction } from 'lodash';
 
-const isModifiedEvent = (event) =>
-  !!(event.metaKey || event.altKey || event.ctrlKey || event.shiftKey);
-
-export class LinkContainer extends Component {
-  static propTypes = {
-    history: PropTypes.shape({
-      push: PropTypes.func.isRequired,
-      replace: PropTypes.func.isRequired,
-      createHref: PropTypes.func.isRequired,
-    }).isRequired,
-    location: PropTypes.object,
-    match: PropTypes.object,
-    staticContext: PropTypes.object,
-    children: PropTypes.element.isRequired,
-    onClick: PropTypes.func,
-    replace: PropTypes.bool,
-    to: PropTypes.oneOfType([
-      PropTypes.string,
-      PropTypes.object,
-    ]).isRequired,
-    exact: PropTypes.bool,
-    strict: PropTypes.bool,
-    className: PropTypes.string,
-    activeClassName: PropTypes.string,
-    style: PropTypes.object,
-    activeStyle: PropTypes.object,
-    isActive: PropTypes.func,
-  };
-
-  static defaultProps = {
-    replace: false,
-    exact: false,
-    strict: false,
-    activeClassName: 'active',
-  };
-
-  handleClick = (event) => {
-    const { children, onClick } = this.props;
+const isModifiedEvent = (event) => !!(event.metaKey || event.altKey || event.ctrlKey || event.shiftKey);
 
+const LinkContainer = ({
+  children,
+  onClick,
+  replace, // eslint-disable-line no-unused-vars
+  to,
+  activeClassName,
+  className,
+  activeStyle,
+  style,
+  isActive: getIsActive,
+  // eslint-disable-next-line comma-dangle
+  ...props
+}) => {
+  const navigate = useNavigate();
+  const href = useHref(typeof to === 'string' ? { pathname: to } : to);
```

Comment:
I think we can just pass `to` here as is?  Looks like `useHref` will parse both object and string forms.